core/domain/classifier_services.py

```diff
@@ -153,6 +153,75 @@ def handle_non_retrainable_states(exploration, state_names, exp_versions_diff):
         job_exploration_mappings)
 
 
+FLOAT_INDICATOR_KEY = 'float_values'
+# pylint: disable=too-many-branches
+
+
+def convert_strings_to_float_numbers_in_classifier_data(classifier_data):
+    """Converts all floating point numbers in classifier data to string.
+
+    Args:
+        classifier_data: dict|list. The trained classifier model in which float
+            values are stored as strings.
+
+    Returns:
+        Dict. Original classifier data dict with float values converted back
+            from string to float.
+    """
+    if isinstance(classifier_data, dict):
+        if not FLOAT_INDICATOR_KEY in classifier_data:
+            # If classifier data does not contain 'float_values' key then
+            # none of its key values need transformation.
+            return classifier_data
+
+        float_fields = classifier_data.pop(FLOAT_INDICATOR_KEY)
+        for k in classifier_data:
+            if isinstance(classifier_data[k], dict):
+                classifier_data[k] = (
+                    convert_strings_to_float_numbers_in_classifier_data(
+                        classifier_data[k]))
+            elif isinstance(classifier_data[k], list):
+                if k in float_fields:
+                    classifier_data[k] = (
+                        convert_strings_to_float_numbers_in_classifier_data(
```

Comment:
Might be better to handle this case inline -- see review on other PR (oppia-ml/26).